include/librealsense2/dds/dds-device.cpp

```diff
@@ -0,0 +1,92 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "dds-device.h"
+#include "dds-participant.h"
+#include "topics/device-info/device-info-msg.h"
+#include <librealsense2/utilities/easylogging/easyloggingpp.h>
+#include <fastdds/rtps/common/Guid.h>
+
+#include <map>
+
+using namespace eprosima::fastdds::dds;
+
+namespace librealsense {
+namespace dds {
+
+
+namespace {
+// The map from guid to device is global
+typedef std::map< dds::dds_guid, std::weak_ptr< dds::dds_device > > guid_to_device_map;
+guid_to_device_map guid_to_device;
+std::mutex devices_mutex;
+}  // namespace
+
+
+class dds_device::impl
+{
+public:
+    topics::device_info _info;
+    std::shared_ptr< dds::dds_participant > const _participant;
+
+    impl( std::shared_ptr< dds::dds_participant > const & participant,
```

Comment:
Why do we need const &? don't you want the participant shared_ptr ref count to increase and hold the participant exist while you have active dds devices?